tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateStockFeatureContext.php

```diff
@@ -0,0 +1,325 @@
+     * @param TableNode $table
+     */
+    public function updateProductStock(string $productReference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+        $productId = $this->getSharedStorage()->get($productReference);
+
+        try {
+            $command = new UpdateProductStockCommand($productId);
+            $unhandledData = $this->setUpdateStockCommandData($data, $command);
+            Assert::assertEmpty(
+                $unhandledData,
+                sprintf('Not all provided data was handled in scenario. Unhandled: %s', var_export($unhandledData, true))
+            );
+            $this->getCommandBus()->handle($command);
+        } catch (ProductException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then product :productReference should have following stock information:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function assertStockInformation(string $productReference, TableNode $table)
+    {
+        $productForEditing = $this->getProductForEditing($productReference);
+        $data = $table->getRowsHash();
+
+        if (isset($data['out_of_stock_type'])) {
+            $data['out_of_stock_type'] = $this->convertOutOfStockToInt($data['out_of_stock_type']);
+        }
+        if (isset($data['pack_stock_type'])) {
+            $data['pack_stock_type'] = $this->convertPackStockTypeToInt($data['pack_stock_type']);
+        }
+
+        $this->assertBoolProperty($productForEditing, $data, 'use_advanced_stock_management');
+        $this->assertBoolProperty($productForEditing, $data, 'depends_on_stock');
+        $this->assertStringProperty($productForEditing, $data, 'pack_stock_type');
+        $this->assertIntegerProperty($productForEditing, $data, 'out_of_stock_type');
+        $this->assertIntegerProperty($productForEditing, $data, 'quantity');
+        $this->assertIntegerProperty($productForEditing, $data, 'minimal_quantity');
+        $this->assertStringProperty($productForEditing, $data, 'location');
+        $this->assertIntegerProperty($productForEditing, $data, 'low_stock_threshold');
+        $this->assertBoolProperty($productForEditing, $data, 'low_stock_alert');
+        $this->assertDateProperty($productForEditing, $data, 'available_date');
+
+        // Assertions checking isset() can hide some errors if it doesn't find array key,
+        // to make sure all provided fields were checked we need to unset every asserted field
+        // and finally, if provided data is not empty, it means there are some unnasserted values left
+        Assert::assertEmpty($data, sprintf('Some provided product stock fields haven\'t been asserted: %s', var_export($data, true)));
+    }
+
+    /**
+     * @Then product :productReference last stock movement has following details:
+     */
+    public function assertProductLastStockMovement(string $productReference, TableNode $table)
```

Comment:
again no big deal, but IMO void lets us now that type is not forgotten and its not intended to return anything on purpose.

```suggestion
    public function assertProductLastStockMovement(string $productReference, TableNode $table): void
```